packages/@o3r/rules-engine/src/engine/debug/engine.debug.ts

```diff
@@ -303,15 +320,7 @@ export class EngineDebugger {
     _id: string,
     factValue$: Observable<any>
   ) {
-    factValue$.subscribe(async () => {
-      const timestamp = Date.now();
-      const facts = await this.getFactsSnapshot(this.registeredRuleEngine!.getRegisteredFactsNames());
-      this.debugEventsSubject$.next(() => ({
-        timestamp,
-        type: 'AvailableFactsSnapshot',
-        facts
-      }));
-    });
+    factValue$.subscribe(() => this.requestFactsSnapshot.next());
```

Comment:
Why not ?
```suggestion
    factValue$.pipe(debounceTime(1000)).subscribe(async () => {
      const timestamp = Date.now();
      const facts = await this.getFactsSnapshot(this.registeredRuleEngine!.getRegisteredFactsNames());
      this.debugEventsSubject$.next(() => ({
        timestamp,
        type: 'AvailableFactsSnapshot',
        facts
      }));
    }));
```
Do we want to be able to trigger `requestFactsSnapshot` in another case ?

Reply:
indeed ! 😃